ocaml/xapi/extauth_plugin_ADwinbind.ml

```diff
@@ -693,6 +695,31 @@ module Winbind = struct
       in
       error "Service not ready error: %s" msg ;
       raise (Auth_service_error (E_GENERIC, msg))
+
+  let random_string len =
+    let upper_char_start = 65 in
+    let upper_char_len = 26 in
+    let rec aux n acc =
+      if n > 0 then
+        aux (n - 1)
+          ((upper_char_start + Random.int upper_char_len |> char_of_int) :: acc)
+      else
+        acc
+    in
+    aux len [] |> List.to_seq |> String.of_seq
```

Comment:
`String.init len (fun _ -> upper_char_start + Random.int upper_char_len |> char_of_int)`

should do the same thing and is a bit simpler